.circleci/config.yml

```diff
@@ -581,70 +581,51 @@ jobs:
-          mkdir stats
-          if [[ $FAILED_COUNT -eq 0 ]]; then
-              echo "success" > stats/outcome.txt
-          else
-              echo "failure" > stats/outcome.txt
-          fi
-          cat stats/outcome.txt
-      - persist_to_workspace:
-            root: /tmp/workspace
-            paths:
-                - repo/stats/
-  send-stats:
-    executor: ubuntu-machine-amd64
-    working_directory: /tmp/workspace/repo
-    steps:
-      - attach_workspace:
-          at: /tmp/workspace
       - run:
-          when: always
-          name: Load start time from file
+          name: Wait for the workflow to complete
           command: |
-            START_TIME=$(cat stats/start-time.txt)
+            # Record the time this step started
+            START_TIME=$(date +%s)
+
+            # wait for the workflow to be done
+            while [[ $(curl --location --request GET "https://circleci.com/api/v2/workflow/$CIRCLE_WORKFLOW_ID/job"| jq -r '.items[]|select(.name != "waiter" and .name != "push" and .name != "report")|.status' | grep -c "running") -gt 0 ]]; do
+              sleep 10
+            done
+
+            # check if a step failed / determine the outcome
+            FAILED_COUNT=$(curl --location --request GET "https://circleci.com/api/v2/workflow/$CIRCLE_WORKFLOW_ID/job" | jq -r '.items[]|.status' | grep -c "failed") || true
+            echo "failed count: $FAILED_COUNT"
+            if [[ $FAILED_COUNT -eq 0 ]]; then
+                OUTCOME="success"
+            else
+                OUTCOME="failure"
+            fi
+            echo "outcome: $OUTCOME"
+
+            # Record the time this step is done
             END_TIME=$(date +%s)
-            OUTCOME=$(cat stats/outcome.txt)
-            echo '{"run_id": "'$CIRCLE_WORKFLOW_ID'", "start": '$START_TIME', "end": '$END_TIME', "commit": "'$CIRCLE_SHA1'", "branch": "'$CIRCLE_BRANCH'", "repository": "'$CIRCLE_PROJECT_USERNAME'/'$CIRCLE_PROJECT_REPONAME'", "outcome": "'$OUTCOME'"}' > stats/stats.json
+
+            # Build the payload
+            echo '{"run_id": "'$CIRCLE_WORKFLOW_ID'", "start": '$START_TIME', "end": '$END_TIME', "commit": "'$CIRCLE_SHA1'", "branch": "'$CIRCLE_BRANCH'", "repository": "'$CIRCLE_PROJECT_USERNAME'/'$CIRCLE_PROJECT_REPONAME'", "outcome": "'$OUTCOME'"}' > stats.json
             echo 'Sending: '$(cat stats/stats.json)
-            curl -X POST "https://api.tinybird.co/v0/events?name=circle_ci_workflows" -H "Authorization: Bearer $TINYBIRD_CI_TOKEN" -d @stats/stats.json
 
+            # Send the data to Tinybird
+            curl -X POST "https://api.tinybird.co/v0/events?name=circle_ci_workflows" -H "Authorization: Bearer $TINYBIRD_CI_TOKEN" -d @stats.json
 
+            # Fail this step depending on the success to trigger a rerun of this step together with others in case of a "rerun failed"
+            [[ $OUTCOME = "success" ]] && exit 0 || exit 1
 
 workflows:
   main:
     jobs:
-      - save-start-time:
+      - waiter:
```

Comment:
Maybe we can rename this into something more descriptive now.

Maybe `push-to-tinybird` like in the GitHub actions so that it's more easily discoverable. 

Need to add this name to the filter in the waiter however though